src/tests/ftest/io/parallel_io.py

```diff
@@ -192,3 +241,139 @@ def test_parallelio(self):
+        # different times and get appended in the self.container variable in
+        # unorderly manner, causing problems during the write process.
+        for _, pool in enumerate(self.pool):
+            for _ in range(self.cont_count):
+                self.create_cont(pool)
+
+        # Try to access each dfuse mounted container using ls. Once it is
+        # accessed successfully, go ahead and perform io on that location
+        # using ior. This process of performing io is done in parallel for
+        # all containers using threads.
+        for pool_count, pool in enumerate(self.pool):
+            dfuse_pool_dir = str(self.dfuse.mount_dir.value + "/" + pool.uuid)
+            for counter in range(self.cont_count):
+                cont_num = (pool_count * self.cont_count) + counter
+                dfuse_cont_dir = str(dfuse_pool_dir + "/" +
+                                     self.container[cont_num].uuid)
+                cmd = u"###ls -a {}".format(dfuse_cont_dir)
+                self.execute_cmd(cmd)
+
+                # run ior on all containers
+                test_file = dfuse_cont_dir + "/testfile"
+                self.ior_cmd.test_file.update(test_file)
+                self.ior_cmd.set_daos_params(
+                    self.server_group, pool, self.container[cont_num].uuid)
+                thread = threading.Thread(
+                    target=self.run_ior,
+                    args=(self.get_ior_job_manager_command(), processes, None,
+                          False))
+                threads.append(thread)
+                thread.start()
+
+        # wait for all ior jobs to be finished
+        for job in threads:
+            job.join()
+
+        # Record free space after io
+        statvfs_before_cont_destroy = self.statvfs_pool(
+            self.dfuse.mount_dir.value)
+
+        # Destroy half of the containers from each pool
+        pfinal = 0
+        for count in range(self.cont_count):
+            pinitial = pfinal
+            pfinal = pinitial + (self.cont_count / 2)
+            del self.container[pinitial:pfinal]
+
+        for cont in self.container:
+            cont_thread = threading.Thread(target=cont.destroy)
+            cont_threads.append(cont_thread)
+            cont_thread.start()
+
+        for destroy_job in cont_threads:
+            destroy_job.join()
+
+        # Record free space after container destroy.
+        statvfs_info_after_cont_destroy = self.statvfs_pool(
+            self.dfuse.mount_dir.value)
+
+        # Calculate the expected space to be returned after containers
+        # are destroyed.
+        reduced_space = (self.cont_count *
+                         int(self.ior_cmd.block_size.value))/2
+
+        # Verify if expected space is returned for each pool after containers
+        # were destroyed. If not, wait for 60 secs and check again. Wait 4
+        # times, otherwise exit the test with a failure.
+        for count in range(self.pool_count):
```

Comment:
In the worst case, if something goes wrong, this for loop is gonna wait for 10 * 60 * 4 seconds. Why not run this as threads for each pool as well ?

Reply:
Done